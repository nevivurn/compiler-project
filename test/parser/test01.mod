//
// test01
//
// parser test:
// - comments
// - simple empty module
//

module test01;

end test01.
