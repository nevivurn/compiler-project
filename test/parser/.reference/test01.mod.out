parsing 'test01.mod'...
successfully parsed.
  AST:
    CAstScope: 'test01'
      symbol table:
        [[
          [ *DIM(<ptr(8) to <NULL>>,<integer>) --> <integer>           ]
          [ *DOFS(<ptr(8) to <NULL>>) --> <integer>           ]
          [ *ReadInt() --> <integer>           ]
          [ *ReadLong() --> <longint>           ]
          [ *WriteChar(<char>) --> <NULL>           ]
          [ *WriteInt(<integer>) --> <NULL>           ]
          [ *WriteLn() --> <NULL>           ]
          [ *WriteLong(<longint>) --> <NULL>           ]
          [ *WriteStr(<ptr(8) to array of <char>>>) --> <NULL>           ]
          [ main     <NULL>           ]
        ]]
      statement list:
        empty.
      nested scopes:
        empty.
    


run the following command to convert the .dot file into a PDF:
  dot -Tpdf -otest01.mod.ast.pdf test01.mod.ast.dot


Done.
